ibis/expr/types/joins.py

```diff
@@ -81,15 +81,15 @@ def dereference_sides(left, right, deref_left, deref_right):
     return left, right
 
 
-def dereference_binop(pred, deref_left, deref_right):
+def dereference_comparison_op(pred, deref_left, deref_right):
     left, right = dereference_sides(pred.left, pred.right, deref_left, deref_right)
     return pred.copy(left=left, right=right)
 
 
 def dereference_value(pred, deref_left, deref_right):
     deref_both = {**deref_left, **deref_right}
-    if isinstance(pred, ops.Binary) and pred.left.relations == pred.right.relations:
-        return dereference_binop(pred, deref_left, deref_right)
+    if isinstance(pred, ops.Comparison) and pred.left.relations == pred.right.relations:
```

Comment:
Can you pull this change out with a failing test case?

Reply:
The test case included here is failing without my change.

Reply:
Done in https://github.com/ibis-project/ibis/pull/7996/